src/main/java/com/github/bpiatek/bbghbackend/model/article/search/ArticleSearcherImpl.java

```diff
@@ -0,0 +1,169 @@
+package com.github.bpiatek.bbghbackend.model.article.search;
+
+import com.github.bpiatek.bbghbackend.model.comment.CommentOpinionStatus;
+import com.github.bpiatek.bbghbackend.util.sql.SqlSelectStatement;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.*;
+import java.math.BigInteger;
+import java.sql.Timestamp;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Błażej Rybarkiewicz <b.rybarkiewicz@gmail.com>
+ */
+@Service
+public class ArticleSearcherImpl implements ArticleSearcher {
+
+  @PersistenceContext
+  private final EntityManager em;
+
+  public ArticleSearcherImpl(EntityManager entityManager) {
+    this.em = entityManager;
+  }
+
+  @Override
+  public List<ArticleSearchFilter> getFiltersFromSearchQuery(String query) {
+    List<ArticleSearchFilter> filterList = new ArrayList<>();
+    if (query == null) {
+      return filterList;
+    }
+    query = query.replaceAll("\\s+","").toLowerCase();
+    Pattern pattern = Pattern.compile("(\\w+?)(:|<|>|=)([^,]+?),");
+    Matcher matcher = pattern.matcher(query + ",");
+    while (matcher.find()) {
+      filterList.add(new ArticleSearchFilter(matcher.group(1), matcher.group(2), matcher.group(3)));
+    }
+    return filterList;
+  }
+
+  @Override
+  public Page<ArticleSearchResult> find(Pageable pageable, List<ArticleSearchFilter> filterList) {
```

Comment:
ta metoda jest mega dluga, ale oczywiscie wazne, ze jest :)
mozna by zrobic np. wyciagniecie tworzenia `Map<String, String> fields` do osobnej metody:

`private Map<String, String> createFieldsMap() {
    Map<String,String> fields = new LinkedHashMap<>();

    String selectCountByStatus = "(SELECT COUNT(id) FROM comment WHERE article_id = article.id AND comment_opinion_status = '%s')";
    // fields are lowercase to simplify filter step
    // alias, field
    fields.put("id", "id");
    fields.put("url", "url");
    fields.put("title", "title");
    fields.put("creationdate", "creation_date");
    fields.put("content", "content");
    fields.put("comments", "(SELECT COUNT(id) FROM comment WHERE article_id = article.id)");
    fields.put("positive", format(selectCountByStatus, POSITIVE));
    fields.put("neutral", format(selectCountByStatus, NEUTRAL));
    fields.put("negative", format(selectCountByStatus, NEGATIVE));
    fields.put("notopinion", format(selectCountByStatus, NOT_OPINION));
    fields.put("notchecked", format(selectCountByStatus, NOT_CHECKED));
    return fields;
  }`
 
to juz bedzie krotsza o 20 wierszy

Reply:
Uwagi o skrócenie wziąłem sobie do serca!